src/framework/results/snapshot.hpp

```diff
@@ -315,6 +248,14 @@ void AverageData::accum_helper(json_t &lhs, json_t &rhs, bool subtract) {
   }
 }
 
+//------------------------------------------------------------------------------
+// JSON serialization
+//------------------------------------------------------------------------------
+
+void to_json(json_t &js, const AverageSnapshot &snapshot) {
```

Comment:
And a `rvalue` overload will add some speedup:
```c++
void to_json(json_t &js, AverageSnapshot&& snapshot) noexcept {
    js = std::move(snapshot);
}
```

Reply:
This file is legacy and removed in next PR